### PR TITLE
Add keepalive method

### PR DESCRIPTION
**Pull request by AudriusButkevicius**

### CODE REVIEW

**Comment by ccding** [comment]
I don't understand why we need to keep alive for UDP listening.

File: stun/client.go [hunk 1]
```diff
@@ -93,3 +94,27 @@ func (c *Client) Discover() (NATType, *Host, error) {
 	}
 	return discover(conn, serverUDPAddr, c.softwareName, c.logger)
 }
+
+// Keepalive sends and receives a bind request, which ensures the mapping stays open
+// Only applicable when client was created with a connection.
+func (c *Client) Keepalive() (*Host, error) {
+	if c.conn == nil {
+		return nil, fmt.Errorf("no connection available")
```
Comment (ccding) [inline stun/client.go]: errors.New("no connection available")?

**Comment by AudriusButkevicius** [comment]
It's to make sure that the mapping on the router is not lost.

**Comment by AudriusButkevicius** [comment]
replaced fmt with errors, yet fmt just calls errors

**Comment by AudriusButkevicius** [comment]
From RFC 5389:

```
It can
   also be used to check connectivity between two endpoints, and as a
   keep-alive protocol to maintain NAT bindings

...

It also provides a way for an endpoint to keep a NAT binding
   alive.

...

 When used in either request/
   response or in indication transactions, the Binding method can also
   be used to keep these "bindings" alive.
```

File: stun/client.go [hunk 2]
```diff
@@ -93,3 +94,27 @@ func (c *Client) Discover() (NATType, *Host, error) {
...
+	}
+	if c.serverAddr == "" {
+		c.SetServerAddr(DefaultServerAddr)
+	}
+	serverUDPAddr, err := net.ResolveUDPAddr("udp", c.serverAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	_, packet, _, _, host, err := test1(c.conn, serverUDPAddr, c.softwareName)
```
Comment (ccding) [inline stun/client.go]: can we call sendChangeReq(...) here?

Comment (AudriusButkevicius) [inline stun/client.go]: I'd have to refactor getting the address to be part of the packet potentially, just to avoid duplication.

Comment (ccding) [inline stun/client.go]: I see. Thanks

**Pull request merged**